app/src/main/java/com/duckduckgo/app/bookmarks/service/SavedSitesImporter.kt

```diff
@@ -0,0 +1,67 @@
+package com.duckduckgo.app.bookmarks.service
+
+import android.content.ContentResolver
+import android.net.Uri
+import com.duckduckgo.app.bookmarks.db.BookmarkEntity
+import com.duckduckgo.app.bookmarks.db.BookmarksDao
+import com.duckduckgo.app.bookmarks.db.FavoriteEntity
+import com.duckduckgo.app.bookmarks.db.FavoritesDao
+import com.duckduckgo.app.bookmarks.model.SavedSite
+import org.jsoup.Jsoup
+
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+interface SavedSitesImporter {
+    suspend fun import(uri: Uri): ImportSavedSitesResult
+}
+
+sealed class ImportSavedSitesResult {
+    data class Success(val savedSites: List<SavedSite>) : ImportSavedSitesResult()
+    data class Error(val exception: Exception) : ImportSavedSitesResult()
+}
+
+class RealSavedSitesImporter(
+    private val contentResolver: ContentResolver,
+    private val bookmarksDao: BookmarksDao,
+    private val favoritesDao: FavoritesDao,
```

Comment:
Same as above, should we be accessing via repository instead of dao?

Reply:
That's a good idea!